core/io/pread_spec.rb

```diff
@@ -26,11 +26,73 @@
       buffer.should == "567"
     end
 
+    it "shrinks the buffer in case of less bytes read" do
+      buffer = "foo"
+      @file.pread(1, 0, buffer)
+      buffer.should == "1"
+    end
+
+    it "grows the buffer in case of more bytes read" do
+      buffer = "foo"
+      @file.pread(5, 0, buffer)
+      buffer.should == "12345"
+    end
+
     it "does not advance the file pointer" do
       @file.pread(4, 0).should == "1234"
       @file.read.should == "1234567890"
     end
 
+    it "returns an empty string for maxlen = 0, regardless of the offset" do
+      @file.pread(0, 4).should == ""
+      @file.pread(0, 400).should == ""
+    end
+
+    it "does not reset the buffer when reading with maxlen = 0" do
+      buffer = "foo"
+      @file.pread(0, 4, buffer)
+      buffer.should == "foo"
+
+      @file.pread(0, 400, buffer)
+      buffer.should == "foo"
+    end
+
+    it "ignores the current offset" do
+      @file.pos = 3
+      @file.pread(4, 0).should == "1234"
+    end
+
+    it "converts maxlen and offset to Integers using #to_int" do
+      maxlen = mock('maxlen')
+      maxlen.should_receive(:to_int).and_return(4)
+      offset = mock('offset')
+      offset.should_receive(:to_int).and_return(0)
+      @file.pread(maxlen, offset).should == "1234"
+    end
+
+    it "converts a buffer to String using to_str (but you cannot use this value)" do
+      buffer = mock('buffer')
+      buffer.should_receive(:to_str).and_return("foo")
+      @file.pread(0, 4, buffer).should == "foo"
+      buffer.should_not.is_a?(String)
+      buffer.to_str.should == "foo"
+    end
```

Comment:
I'm not really sure about this one: this is the current behaviour of MRI, but this feels like an implementation detail/bug, since the value cannot be used.

Reply:
The behaviour appears to be different between Linux and Windows as well.